mesonbuild/modules/hotdoc.py

```diff
@@ -338,9 +338,18 @@ def make_targets(self):
 
         install_script = None
         if install:
+            datadir = os.path.join(self.state.get_option('prefix'), self.state.get_option('datadir'))
+            if self.kwargs.get('devhelp_activate', False):
```

Comment:
What happens if a silly meson.build sets this to something other than a boolean? Does hotdoc itself error out?

Reply:
Those extra kwargs are not really checked, it's probably not going to end well in https://github.com/mesonbuild/meson/blob/master/mesonbuild/modules/hotdoc.py#L81

Reply:
Well yeah, that's kind of the thing. Extra values are up to the user to get correct, and they're fully responsible for getting that right, or dealing with the fallout if they don't.

But for anything we explicitly check it's often nicer to use KwargInfo, as long as we're guaranteed that post-1.0 it doesn't break existing meson.build files no matter how silly.

So if passing a string here was always an error, it costs nothing to make this a KwargInfo, guaranteeing sensible behavior and avoiding the need to `.get()` while burying the fallback value in the depths of the logic.

Reply:
You're right, let's validate that kwarg, done.

Reply:
I double checked. Passing a string is an error, it results in `hotdoc --devhelp-activate=thestring`, passing anything that is neither a string nor a boolean will silently work because it results in `hotdoc --devhelp-activate otherarg` and hotdoc discards the unknown argument.

So we do need to accept any type, and likely issue a deprecation warning.

Reply:
I don't think it's worth going all the way, there is literally only gst and meson using hotdoc. We added type checking in many places without deprecation in the past.

Reply:
Yes, before 1.0

@dcbaker

Reply:
1.0 is just a number, there are no more nor less backward compatibility guarantee. Anyway, made the change.